dashboard/db/migrate/20240122193916_create_ai_tutor_interactions.rb

```diff
@@ -0,0 +1,20 @@
+class CreateAiTutorInteractions < ActiveRecord::Migration[6.1]
+  def change
+    create_table :ai_tutor_interactions do |t|
+      t.integer :user_id, null: false, foreign_key: true
+      t.integer :level_id, foreign_key: true
+      t.integer :script_id, foreign_key: true
+      t.string :ai_model_version
+      t.string :type
+      t.string :project_id
+      t.string :project_version_id
+      t.text :prompt
+      t.string :status
+      t.text :ai_response
+
+      t.timestamps
+    end
+
+    add_index :ai_tutor_interactions, [:user_id, :level_id, :script_id], unique: false, name: "index_ati_user_level_script"
```

Comment:
TIL index names have a max of 64 characters, but you can supply your own shorter name if the auto generated one is too long.

Reply:
Do you want any individual indexes, like on `user_id`, today? Adding an index to a large table can be a costly operation, especially if the column contains diverse data like that one will. Today's use case seems like it will only use the composite index, but the future use case of 'all for the given user' seems incredibly likely, if not for features, than for backend investigation & troubleshooting.

Reply:
oh, good idea! For some reason I was thinking of a composite index OR individual indices, but having both makes sense. Thank you.